cdap-ui/app/features/tracker/controllers/integrations-ctrl.js

```diff
@@ -29,15 +29,35 @@ class TrackerIntegrationsController {
     this.navigatorSetup = {
       isOpen: false,
       isSetup: false,
-      isEnabled: false
+      isEnabled: false,
+      popoverEnabled: true
     };
     this.showConfig = false;
 
+    this.optionalSettings = {
+      navigator: false,
+      kafka: false
+    };
+
     this.navigatorInfo = {
-      brokerString: '',
-      hostname: '',
-      username: '',
-      password: ''
+      navigatorConfig: {
```

Comment:
Should we get this from a configuration factory just like we do in dag directive. It might be easier to plugin it in another factory in case if we need to integrate other management tools. Just a suggestion, if the changes are too big then we can move it to another PR. We name this file as `integrations-ctrl.js` but the implementation is too specific for navigator.